packages/web/src/pages/oauth-login-page/OAuthLoginPage.tsx

```diff
@@ -267,151 +311,190 @@ export const OAuthLoginPage = () => {
-              />
-              {signInError == null ? null : (
-                <div className={styles.credentialsErrorContainer}>
-                  <IconValidationX
-                    width={14}
-                    height={14}
-                    className={styles.credentialsErrorIcon}
+          </Flex>
+          {userAlreadyWriteAuthorized ? null : (
+            <PermissionsSection
+              scope={scope}
+              tx={tx as WriteOnceTx}
+              userEmail={isInManagerMode ? userEmail : null}
+              isLoggedIn={isLoggedIn}
+              isLoading={userEmail === null}
+              txParams={txParams}
+            />
+          )}
+          <div className={styles.formArea}>
+            {isLoggedIn ? (
+              <div className={styles.userInfoContainer}>
+                <Text
+                  variant='body'
+                  size='m'
+                  css={{ color: 'var(--harmony-n-600)' }}
+                >
+                  {messages.signedInAs}
+                </Text>
+                <div className={styles.tile}>
+                  <ProfileInfo
+                    displayNameClassName={styles.userInfoDisplayName}
+                    handleClassName={styles.userInfoHandle}
+                    centered={false}
+                    imgClassName={styles.profileImg}
+                    className={styles.userInfo}
+                    user={account}
                   />
-                  <span className={styles.errorText}>{signInError}</span>
                 </div>
-              )}
-              {showOtpInput ? (
-                <Input
-                  placeholder='Verification Code'
-                  size='medium'
-                  name='otp'
-                  value={otpInput}
-                  characterLimit={6}
-                  type='number'
-                  variant={'normal'}
-                  onChange={setOtpInput}
-                  className={cn(styles.otpInput)}
-                />
-              ) : null}
-              <CTAButton type='submit' isLoading={isSubmitting}>
-                {messages.signInButton}
-              </CTAButton>
-            </form>
-            <div className={styles.signUpButtonContainer}>
-              <a
-                className={styles.linkButton}
-                href={SIGN_UP_PAGE}
-                target='_blank'
-                rel='noopener noreferrer'
-              >
-                {messages.signUp}
-              </a>
-            </div>
-          </div>
-        )}
-        {generalSubmitError == null ? null : (
-          <div className={styles.generalErrorContainer}>
-            <span className={styles.errorText}>{generalSubmitError}</span>
+                <Flex mt='l' alignItems='center' justifyContent='space-between'>
+                  <TextLink variant='visible' size='s' onClick={handleSignOut}>
+                    {messages.signOut}
+                  </TextLink>
+                  {accounts.length > 0 ? (
```

Comment:
Probably should be checking the feature flag somewhere in this component? I know it's not possible for the logic which shows the switcher to be true unless somebody has already been in Manager Mode and linked up accounts. But in case we need to turn it off for some reason...